src/test/qtumtests/x86_tests.cpp

```diff
@@ -141,4 +141,58 @@ BOOST_AUTO_TEST_CASE(x86_hypervisor_storage){
 
 }
 
+BOOST_AUTO_TEST_CASE(x86_hypervisor_sha256) {
+    //in theory it's ok to not have a backing database for the wrapper
+    //as long as we don't access a non-existent key or try to commit to database
+    FakeVMContainer fake;
+
+    unsigned char testValue[] = "hello world";
+    fake.cpu.WriteMemory(0x1000, sizeof(testValue), &testValue);
+
+    fake.cpu.SetReg32(EAX, QSC_SHA256);      //syscall number
+    fake.cpu.SetReg32(EBX, 0x1000);            //input location
+    fake.cpu.SetReg32(ECX, sizeof(testValue)); //input size
+    fake.cpu.SetReg32(EDX, 0x1100); // output location
+
+    fake.hv.HandleInt(QtumSystem, fake.cpu);
+    std::string expectedHashVal = "b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9";
+    unsigned char *gHashVal = new unsigned char[32];
```

Comment:
It's test code so not critical at all, but stack allocation (ie, without new) would be preferred here for speed since size is known and it won't be too large for the stack

Reply:
can do.